src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -379,6 +379,13 @@ void ImageReplayer<I>::handle_bootstrap(int r) {
     m_bootstrap_request = nullptr;
   }
 
+  // set m_delete_requested early to ensure that in case remote
+  // image no longer exists local image gets deleted even if start
+  // is canceled
+  if (r == -ENOLINK) {
+    dout(5) << "remote image no longer exists" << dendl;
+    m_delete_requested = true;
```

Comment:
Since `m_delete_requested` is set here, there is no need to set it again below.  Please remove that statement.